packages/kbn-es-types/src/index.ts

```diff
@@ -26,6 +27,7 @@ import {
 
 export type ESFilter = estypes.QueryDslQueryContainer;
 export type ESSearchRequest = estypes.SearchRequest;
+export type ESSearchRequestWithoutBody = estypesWithoutBody.SearchRequest;
```

Comment:
nit: I'd love it if we could start highlighting that `body` is deprecated:

```suggestion
/** @deprecated Use ESSearchRequestWithoutBody instead */
export type ESSearchRequest = estypes.SearchRequest;
export type ESSearchRequestWithoutBody = estypesWithoutBody.SearchRequest;
```

In any case, I think this is a good change... as we'll be able to progressively migrate consumers of each type.